sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeActionsOperation.cs

```diff
@@ -127,7 +134,19 @@ public class AnalyzeActionsOperation : PageableOperation<AnalyzeActionsResult>,
         /// <param name="client">The client used to check for completion.</param>
         public AnalyzeActionsOperation(string operationId, TextAnalyticsClient client)
         {
-            // TODO: Add argument validation here.
+            try
+            {
+                string plainOperationId = ClientCommon.DecodeOperationId(operationId);
+                OperationIdInformation idInformation = JsonSerializer.Deserialize<OperationIdInformation>(plainOperationId);
+                _jobId = idInformation.JobId;
+                _showStats = idInformation.ShowStats;
+                _idToIndexMap = idInformation.InputDocumentOrder;
```

Comment:
Do we need to validate that these are present?

Reply:
I think validation is not necessary. The three properties are set by us when generating the base 64 ID, so I can't think of any real scenario where they would be missing.

An extremally bored user who knows how our code works could figure out a way to pass an invalid operation ID and get some null ref exceptions, but I don't think we should worry about that.

Reply:
I'm not super worried about bored users.  I'm much more worried about future us though.  What if:

1. Someone builds an app that kicks off the operations in one worker and throws them in another queue for someone else to process when they're done
2. Another worker pulls IDs from the queue, rehydrates, and checks if they've finished yet
3. TA LROs add new properties and make required values like `ShowStats` optional in the future
4. The customer updates Step 1 to use a newer client, but step 2 is still using an older client.
5. \*Something Breaks\*

What kind of error would that customer get in Step 2 of their distributed app?  A null reference somewhere?  Broken code because we send a null value for a property to the service?  How easy would it be to debug?

I think we could make it very easy by validating our expectations in `OperationIdInformation.Decode` and throwing an `ArgumentException` if they're not met.

Reply:
> I think we could make it very easy by validating our expectations in OperationIdInformation.Decode and throwing an ArgumentException if they're not met.

Done.